pkg/images/pull.go

```diff
@@ -126,23 +126,20 @@ func handleTarObject(ctx context.Context, tr *tar.Reader, hdr *tar.Header, conf
 		// Copy the target file out to the host (compressed or not).
 		// On failure, clean up the temporary file; otherwise move it
 		// to the target destination.
-		tmpFile, err := os.CreateTemp("", filepath.Base(hdr.Name))
```

Comment:
Woops, forgot to update the comment. Let me do that.

Reply:
```
diff --git a/pkg/images/pull.go b/pkg/images/pull.go
index dc3ff8416db5..1112fc0edf9e 100644
--- a/pkg/images/pull.go
+++ b/pkg/images/pull.go
@@ -124,16 +124,16 @@ func handleTarObject(ctx context.Context, tr *tar.Reader, hdr *tar.Header, conf
                compressed := strings.HasSuffix(dstPath, ".zst")

                // Copy the target file out to the host (compressed or not).
-               // On failure, clean up the temporary file; otherwise move it
-               // to the target destination.
                dstFile, err := os.Create(dstPath)
                if err != nil {
-                       return image, fmt.Errorf("failed to open temporary file for %s: %w", hdr.Name, err)
+                       return image, fmt.Errorf("failed to open %s: %w", dstPath, err)
                }
                defer func() {
                        path := dstFile.Name()
                        dstFile.Close()
                        if err != nil || (!conf.Cache && compressed) {
+                               // Only keep the compressed copy on the hostfs
+                               // if asked for it, and if there are no errors.
                                os.Remove(path)
                                return
                        }
```